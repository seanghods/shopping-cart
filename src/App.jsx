import {
  BrowserRouter as Router,
  Route,
  Routes,
  Navigate,
  useNavigate,
} from 'react-router-dom';
import { useState, useEffect } from 'react';
import { Home, Shop, Cart, NotFound } from './containers';
import NavBar from './components/NavBar';
import './App.css';

function App() {
  const [cart, setCart] = useState([]);
  const [items, setItems] = useState([]);
  const navigate = useNavigate();
  useEffect(() => {
    async function fetchItems() {
      const response = await fetch('https://fakestoreapi.com/products/');
      const data = await response.json();
      setItems(data);
    }
    fetchItems();
  }, []);
  function navigateShop() {
    navigate('/shop');
  }
  function addToCart(e, item) {
    const newCart = [...cart];
    newCart.push(item);
    setCart(newCart);
  }
  function removeFromCart(e) {
    const index = e.target.id;
    const newCart = [...cart];
    newCart.splice(index, 1);
    setCart(newCart);
  }
  return (
    <>
      <NavBar cart={cart} />
      <div className="flex-1">
        <Routes>
          <Route path="/home" element={<Home navigateShop={navigateShop} />} />
          <Route
            path="/shop"
            element={<Shop items={items} addToCart={addToCart} />}
          />
          <Route
            path="/cart"
            element={
              <Cart
                cart={cart}
                removeFromCart={removeFromCart}
                navigateShop={navigateShop}
              />
            }
          />
          <Route path="/" element={<Navigate to="/home" replace />} />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </div>
    </>
  );
}

export default App;
