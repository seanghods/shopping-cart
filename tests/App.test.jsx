import { describe, it, expect } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import App from '../src/App';
// import { vi } from 'vitest';
import { render } from '@testing-library/react';

describe('App component', () => {
  it('renders correctly', () => {
    const container = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
