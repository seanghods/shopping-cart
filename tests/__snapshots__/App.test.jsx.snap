// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nav-bar-section h-screen flex flex-col w-[100px] md:w-[350px] sticky top-0"
      >
        <div
          class="nav-bar flex-1 m-3 md:m-7 bg-[#20639B] rounded-lg text-white shadow-xl shadow-[#106ae0]"
        >
          <div
            class="logo flex flex-col items-center p-3 md:p-11 text-4xl font-bold font-bobsburgers text-center"
          >
            The Sean Shop
            <img
              class="rounded-full w-[50px] md:w-[150px] shadow-xl shadow-white"
              src="/src/assets/sean-shop-pic.png"
            />
          </div>
          <div
            class="nav-bar-links p-11 flex flex-col gap-20 items-center md:items-stretch "
          >
            <a
              aria-current="page"
              class="flex gap-2 text-md hover:font-bold font-bold"
              href="/home"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <p
                class="hidden md:block"
              >
                Home
              </p>
            </a>
            <a
              class="flex gap-2 text-md hover:font-bold "
              href="/shop"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="StoreMallDirectoryOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18.36 9 .6 3H5.04l.6-3h12.72M20 4H4v2h16V4zm0 3H4l-1 5v2h1v6h10v-6h4v6h2v-6h1v-2l-1-5zM6 18v-4h6v4H6z"
                />
              </svg>
              <p
                class="hidden md:block"
              >
                Shop
              </p>
            </a>
            <a
              class="flex gap-2 text-md hover:font-bold "
              href="/cart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShoppingCartOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <p
                class="hidden md:block"
              >
                Cart - 
                0
              </p>
            </a>
          </div>
        </div>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col items-center text-center gap-10 md:gap-20"
        >
          <div
            class="title text-5xl mt-14 m-3 md:m-14 font-bobsburgers"
          >
            <h1>
              Welcome to The Sean Shop
            </h1>
          </div>
          <p
            class="text-2xl font-bobsburgers"
          >
            The one stop shop for all your possible needs!
          </p>
          <img
            alt="fam pic"
            class="w-[1100px]]"
            src="/src/assets/homepage.png"
          />
          <button
            class="bg-[#3CAEA3] tracking-widest rounded-md px-10 py-5 text-white font-bobsburgers text-2xl shadow-md animate-bounce transform transition duration-500 hover:scale-110"
          >
            Shop Now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nav-bar-section h-screen flex flex-col w-[100px] md:w-[350px] sticky top-0"
    >
      <div
        class="nav-bar flex-1 m-3 md:m-7 bg-[#20639B] rounded-lg text-white shadow-xl shadow-[#106ae0]"
      >
        <div
          class="logo flex flex-col items-center p-3 md:p-11 text-4xl font-bold font-bobsburgers text-center"
        >
          The Sean Shop
          <img
            class="rounded-full w-[50px] md:w-[150px] shadow-xl shadow-white"
            src="/src/assets/sean-shop-pic.png"
          />
        </div>
        <div
          class="nav-bar-links p-11 flex flex-col gap-20 items-center md:items-stretch "
        >
          <a
            aria-current="page"
            class="flex gap-2 text-md hover:font-bold font-bold"
            href="/home"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Home
            </p>
          </a>
          <a
            class="flex gap-2 text-md hover:font-bold "
            href="/shop"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StoreMallDirectoryOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m18.36 9 .6 3H5.04l.6-3h12.72M20 4H4v2h16V4zm0 3H4l-1 5v2h1v6h10v-6h4v6h2v-6h1v-2l-1-5zM6 18v-4h6v4H6z"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Shop
            </p>
          </a>
          <a
            class="flex gap-2 text-md hover:font-bold "
            href="/cart"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <p
              class="hidden md:block"
            >
              Cart - 
              0
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="flex-1"
    >
      <div
        class="flex flex-col items-center text-center gap-10 md:gap-20"
      >
        <div
          class="title text-5xl mt-14 m-3 md:m-14 font-bobsburgers"
        >
          <h1>
            Welcome to The Sean Shop
          </h1>
        </div>
        <p
          class="text-2xl font-bobsburgers"
        >
          The one stop shop for all your possible needs!
        </p>
        <img
          alt="fam pic"
          class="w-[1100px]]"
          src="/src/assets/homepage.png"
        />
        <button
          class="bg-[#3CAEA3] tracking-widest rounded-md px-10 py-5 text-white font-bobsburgers text-2xl shadow-md animate-bounce transform transition duration-500 hover:scale-110"
        >
          Shop Now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
